ui/src/app/views/not-found/component.ts

```diff
@@ -0,0 +1,42 @@
+import {Location} from '@angular/common';
+import {Component, Injectable} from '@angular/core';
+import {NavigationError, Router} from '@angular/router';
+
+// Mostly borrowed from the Job Manager.
+@Component({
+    selector: 'app-not-found',
+    templateUrl: './component.html',
+    styleUrls: ['./component.css']
+})
+@Injectable()
```

Comment:
I'm sorry, in my initial look the diff formatting kind of obscured the fact that the "service" and the "component" are the same thing.

I don't know how yet but I bet this leads to issues.  It seems like Angular's DI system is oddly permissive (I've also seen a component that was marked as a module, and there's that business where we are or were directly injecting AppComponent) but I don't see this behavior endorsed or documented anywhere and I'm willing to bet it works by accident and shouldn't be relied on.

Actually why does this component need `Injectable`? Won't it work the same without listing it as a provider?